Tools/CMake/AMReXOptions.cmake

```diff
@@ -110,18 +110,23 @@ if (NOT AMReX_GPU_BACKEND STREQUAL NONE)
    message( STATUS "   AMReX_GPU_BACKEND = ${AMReX_GPU_BACKEND}")
 endif ()
 
+# Legacy variables for internal use only
 if (AMReX_GPU_BACKEND STREQUAL SYCL)
-   set(AMReX_CUDA  OFF CACHE INTERNAL "")
-   set(AMReX_DPCPP ON  CACHE INTERNAL "")
-   set(AMReX_HIP   OFF CACHE INTERNAL "")
+   set(AMReX_DPCPP ON )
+   set(AMReX_CUDA  OFF)
+   set(AMReX_HIP   OFF)
 elseif (AMReX_GPU_BACKEND STREQUAL CUDA)
-   set(AMReX_CUDA  ON  CACHE INTERNAL "")
-   set(AMReX_DPCPP OFF CACHE INTERNAL "")
-   set(AMReX_HIP   OFF CACHE INTERNAL "")
+   set(AMReX_DPCPP OFF)
+   set(AMReX_CUDA  ON )
+   set(AMReX_HIP   OFF)
 elseif (AMReX_GPU_BACKEND STREQUAL HIP)
-   set(AMReX_CUDA  OFF CACHE INTERNAL "")
-   set(AMReX_DPCPP OFF CACHE INTERNAL "")
-   set(AMReX_HIP   ON  CACHE INTERNAL "")
+   set(AMReX_DPCPP OFF)
+   set(AMReX_CUDA  OFF)
+   set(AMReX_HIP   ON )
+else ()
+   set(AMReX_DPCPP OFF)
+   set(AMReX_CUDA  OFF)
+   set(AMReX_HIP   OFF)
```

Comment:
If you set this on `else ()`, will we be able to still set it in legacy usage?

Reply:
I am not sure about legacy usage, but the way it was done before (no extra ``else()``) was too risky. Therefore I am inclined to keep it as is. Downstream codes are not supposed to use those variables anyways.

Reply:
This would deprecate too fast for us, we still use those vars for the superbuild setup.

Reply:
Can you check whether this change impacts your superbuild? I'd keep it rather than wait and eventually forget it

Reply:
Did a first check and this is breaking: https://github.com/ax3l/WarpX/actions/runs/424382804
Will adjust now and report back.

Reply:
@mic84 This will break for AMR-Wind also when using superbuild mode.

Reply:
Okili dokili, WarpX (+1 one other code) updated, e.g. https://github.com/ECP-WarpX/WarpX/pull/1577 :)

Reply:
@sayerhs please let us know when AMR-Wind is good to go

Reply:
discussed offline: AMR-Wind pulls a specific release via submodules and @sayerhs confirms we can go ahead.